### PR TITLE
Fix/validator

### PR DESCRIPTION
**Pull request by Raid5594**
- fix bug whereby all types of logs where validated
- fix bug whereby incorrect events where deposited while setting reward points more than once for specific node & era
- fix bug whereby persistent local storage was not working as expected

### CODE REVIEW

File: pallets/ddc-validator/src/lib.rs [hunk 1]
```diff
@@ -73,6 +73,8 @@ type ResultStr<T> = Result<T, &'static str>;
 /// Offchain local storage key that holds the last era in which the validator completed its
 /// assignment.
 const LAST_VALIDATED_ERA_KEY: &[u8; 40] = b"pallet-ddc-validator::last_validated_era";
+/// Offchain local storage that holds the validation lock
+const VALIDATION_LOCK: &[u8; 40] = b"pallet-ddc-validator::validation_lock";
```
Comment (khssnv) [inline pallets/ddc-validator/src/lib.rs:77]: The type is `&[u8; 37]` here.

File: pallets/ddc-validator/src/dac.rs [hunk 2]
```diff
@@ -216,10 +216,13 @@ pub fn get_acknowledged_bytes_bucket<'a>(
 		let mut total_bytes_received = 0u64;
 		let bucket_id = file_request.bucket_id;
 		for (_, chunk) in &file_request.chunks {
-			if let Some(ack) = &chunk.ack {
-				total_bytes_received += ack.bytes_received;
-			} else {
-				total_bytes_received += get_proved_delivered_bytes(chunk, &ack_timestamps);
+			// Only check reads
+			if chunk.log.log_type == 1u64 {
```
Comment (khssnv) [inline pallets/ddc-validator/src/dac.rs:220]: Would be nice to decode integer type values to enum variants somewhere and `match` on it if possible

File: pallets/ddc-validator/src/lib.rs [hunk 3]
```diff
@@ -86,6 +88,26 @@ pub const DEFAULT_DATA_PROVIDER_URL: &str = "http://webdis:7379";
...
 pub const QUORUM_SIZE: usize = 1;
 
+#[derive(PartialEq, Eq, Clone, Encode, Decode, RuntimeDebug, TypeInfo)]
+pub enum LogType {
+	Read = 1,
+	Write = 2,
+	Query = 3,
+}
+
+impl TryFrom<u64> for LogType {
```
Comment (khssnv) [inline pallets/ddc-validator/src/lib.rs:98]: `FromStr` could be a more explicit choice.

https://paritytech.github.io/polkadot-sdk/master/sp_std/str/trait.FromStr.html

File: pallets/ddc-validator/src/dac.rs [hunk 4]
```diff
@@ -216,10 +216,15 @@ pub fn get_acknowledged_bytes_bucket<'a>(
 		let mut total_bytes_received = 0u64;
 		let bucket_id = file_request.bucket_id;
 		for (_, chunk) in &file_request.chunks {
-			if let Some(ack) = &chunk.ack {
-				total_bytes_received += ack.bytes_received;
-			} else {
-				total_bytes_received += get_proved_delivered_bytes(chunk, &ack_timestamps);
+			// Only check reads
+			match chunk.log.log_type.try_into() {
+				Ok(LogType::Read) =>
```
Comment (khssnv) [inline pallets/ddc-validator/src/dac.rs:221]: :+1:

File: pallets/ddc-validator/src/lib.rs [hunk 5]
```diff
@@ -86,6 +88,26 @@ pub const DEFAULT_DATA_PROVIDER_URL: &str = "http://webdis:7379";
 pub const DATA_PROVIDER_URL_KEY: &[u8; 32] = b"ddc-validator::data-provider-url";
 pub const QUORUM_SIZE: usize = 1;
 
+#[derive(PartialEq, Eq, Clone, Encode, Decode, RuntimeDebug, TypeInfo)]
+pub enum LogType {
+	Read = 1,
+	Write = 2,
+	Query = 3,
```
[the remark this answers is not in the document]
Comment (Raid5594) [reply pallets/ddc-validator/src/lib.rs:95]: Yes, it was updated

File: pallets/ddc-validator/src/lib.rs [hunk 6]
```diff
@@ -86,6 +88,26 @@ pub const DEFAULT_DATA_PROVIDER_URL: &str = "http://webdis:7379";
 pub const DATA_PROVIDER_URL_KEY: &[u8; 32] = b"ddc-validator::data-provider-url";
 pub const QUORUM_SIZE: usize = 1;
 
+#[derive(PartialEq, Eq, Clone, Encode, Decode, RuntimeDebug, TypeInfo)]
+pub enum LogType {
```
[the remark this answers is not in the document]
Comment (Raid5594) [reply pallets/ddc-validator/src/lib.rs:92]: Let's use opCode as it is in updated documentation:
https://www.notion.so/cere/DAC-Storing-data-and-aggregations-approach-v3-22f77c9336d947cbb99566571a00d391?pvs=4#cce948134a984cea8050e5f346df0915

File: pallets/ddc-validator/src/lib.rs (hunk 5 above)
[the remark this answers is not in the document]
Comment (khssnv) [reply pallets/ddc-validator/src/lib.rs:95]: Sorry, I don't see it.

File: pallets/ddc-validator/src/lib.rs (hunk 6 above)
[the remark this answers is not in the document]
Comment (khssnv) [reply pallets/ddc-validator/src/lib.rs:92]: Ok. Let's pass it here as it is and I'll raise a question on renaming in the whole system later.

File: pallets/ddc-validator/src/lib.rs [hunk 7]
```diff
@@ -86,6 +88,26 @@ pub const DEFAULT_DATA_PROVIDER_URL: &str = "http://webdis:7379";
 pub const DATA_PROVIDER_URL_KEY: &[u8; 32] = b"ddc-validator::data-provider-url";
 pub const QUORUM_SIZE: usize = 1;
 
+#[derive(PartialEq, Eq, Clone, Encode, Decode, RuntimeDebug, TypeInfo)]
+pub enum opCode {
```
Comment (khssnv) [inline pallets/ddc-validator/src/lib.rs:92]: `OpCode` please.

**Review by khssnv — APPROVED** [review]

**Pull request merged**